src/engine/guidance/post_processing.cpp

```diff
@@ -60,7 +60,8 @@ inline bool choiceless(const RouteStep &step, const RouteStep &previous)
                                                    step.intersections.front().entry.end(),
                                                    true);
 
-    return is_without_choice;
+    return is_without_choice && step.maneuver.instruction.type != TurnType::EndOfRoad;
```

Comment:
Shouldn't this be good enough to fix the issue we're seeing? What's going in below?